src/completions.rs

```diff
@@ -15,27 +15,30 @@ pub struct Completions<T: LLMModel> {
     model: T,
     //For prompt & response
     max_tokens: usize,
-    temperature: u32,
+    temperature: f32,
     input_json: Option<String>,
     debug: bool,
     function_call: bool,
     api_key: String,
 }
 
 impl<T: LLMModel> Completions<T> {
-    ///
+    /// Constructor for the Completions API
     pub fn new(
         model: T,
         api_key: &str,
         max_tokens: Option<usize>,
         temperature: Option<u32>,
     ) -> Self {
+        let temperature = temperature
+            .map(|temp| model.get_normalized_temperature(temp))
```

Comment:
This way we don't have to modify the constructor, making the change backwards compatible. We still pass an `u32` integer that should represent a 0-100[%] of the range we want to use. The `get_normalized_temperature` exposed by each model normalizes that to the range specific to the model.